webapp/publisher/snaps/views.py

```diff
@@ -806,6 +806,33 @@ def post_register_name():
     return flask.redirect(flask.url_for("account.get_account"))
 
 
+@publisher_snaps.route("/register-snap/json", methods=["POST"])
+@login_required
+def post_register_name_json():
+    snap_name = flask.request.form.get("snap-name")
+
+    if not snap_name:
+        return flask.jsonify({"success": False})
+
+    try:
+        response = api.post_register_name(
+            session=flask.session, snap_name=snap_name
+        )
+    except ApiResponseErrorList as api_response_error_list:
+        return (
+            flask.jsonify(
+                {"success": False, "errors": api_response_error_list.errors}
+            ),
+            api_response_error_list.status_code,
+        )
+    except ApiError as api_error:
+        return _handle_errors(api_error)
```

Comment:
I would just `return flask.jsonify({"errors": [error]}), 500`, for consistency of always returning JSON.

Reply:
@jpmartinspt Doesn't `_handle_errors` do something more (detecting kind of error etc) that we would want to keep?

Reply:
@jpmartinspt I tried to remove `_handle_errors` and just return JSON, but the tests fails when I do that. I think our `BaseTestCases.EndpointLoggedIn` assumes `_handle_errors` is used and has some tests that assure it's in place (to check expired macaroons, etc).

I agree that for JSON endpoints it would be good to have JSON error handling, so maybe that's something we could improve later: having a "JSON equivalent" of `_handle_errors` and/or `BaseTestCases.EndpointLoggedIn`.

Reply:
Yeah I had a similar issue with the metrics endpoint ended up not using the `BaseTestCases.EndpointLoggedIn`.
I think you are right, we might need an `_handle_errors_json`, I'm happy if you leave handle_errors for now, but maybe we could raise a ticket to look into this.